src/Elmish.WPF/BindingVmHelpers.fs

```diff
@@ -618,24 +622,25 @@ type Update
           this.Recursive(b.Get currentModel, b.Get newModel, b.Dispatch, b.Binding)
 
 
-type Get(nameChain: string) =
+type Get<'a>(nameChain: string) =
 
-  member _.Base (model: 'model, binding: BaseVmBinding<'model, 'msg>) =
+  member _.Base (model: 'model, binding: BaseVmBinding<'model, 'msg, 'a>): Result<'a, GetError> =
     match binding with
     | OneWay { OneWayData = d } -> d.Get model |> Ok
     | TwoWay b -> b.Get model |> Ok
     | OneWayToSource _ -> GetError.OneWayToSource |> Error
-    | OneWaySeq { Values = vals } -> vals.BoxedCollection() |> Ok
-    | Cmd cmd -> cmd |> box |> Ok
-    | SubModel { Vm = vm } -> vm.Value |> ValueOption.toObj |> box |> Ok
+    | OneWaySeq { Values = vals } -> vals.BoxedCollection() |> unbox |> Ok
```

Comment:
Can we add a type parameter to `CollectionTarget` so that we don't have to call `unbox` here?

Reply:
This PR and PR #473 are similar (they are adding type parameters) while PR #472 improves code organization.

Can the order of these changes be grouped so that all the code organization improvements happen first and then the changes to add type parameters?

Reply:
I was avoiding it because of the code conflicts that would entail, but I think you're right. I'll try to reorder it.

Reply:
I combined this PR with #473

Reply:
Also removed the `unbox` like should have happened before.